pkg/mcproducer/mcproducer.go

```diff
@@ -52,29 +46,30 @@ func ProduceMachineConfig(machineConfigName,
 		workerImageToUse = workerImage
 	}
 
-	templateParams := map[string]any{
-		"FirmwareFilesPath":         firmwareFilesPath,
-		"KernelModuleImage":         kernelModuleImage,
-		"KernelModule":              kernelModuleName,
-		"MachineConfigPoolRef":      machineConfigPoolRef,
-		"MachineConfigName":         machineConfigName,
-		"KernelModuleImageFilepath": kernelModuleImageFilepath,
-		"InTreeModuleToRemove":      inTreeModuleToRemove,
-		"WorkerImage":               workerImageToUse,
-		"WorkerConfigFilepath":      workerConfigFilepath,
+	mcfgAPI := mcfg.NewMCFG()
+	jsonIgnition, err := mcfgAPI.GenerateIgnition(kernelModuleImage, kernelModuleName, inTreeModuleToRemove, firmwareFilesPath, workerImageToUse, machineConfigName)
+	if err != nil {
+		return "", fmt.Errorf("failed to create ignition string: %v", err)
 	}
 
-	templateParams["ReplaceInTreeDriverContents"] = base64.StdEncoding.EncodeToString([]byte(scriptReplaceKmod))
-	templateParams["PullKernelModuleContents"] = base64.StdEncoding.EncodeToString([]byte(scriptPullImage))
-	templateParams["WaitForNetworkDispatcherContents"] = base64.StdEncoding.EncodeToString([]byte(scriptWaitForNetworkDispatcher))
+	yamlIgnition, err := yaml.JSONToYAML(jsonIgnition)
```

Comment:
The mcfgAPI is converting the yaml ignition to a json ignition and here we are re-converting. Can we just return both json and yaml ignitions from the mcfg API? Or don't convert to json to begin with in the mcfg API?